### PR TITLE
feat: curated audit remediations (SLSA caller, pre-commit CI gate, operator-script hardening)

### PR DESCRIPTION
**Pull request by williaby**
## What this is

A curated subset of `claude/repo-audit-oToMV`. That branch bundled a repo-wide
audit plus remediations; this PR keeps only the remediations that were verified
correct, drops the parts built on a stale premise, and keeps the audit findings
manifest as a local gitignored file rather than publishing the audit docs.

## Included (verified)

- **`fix`: SLSA template caller + FIPS doc path (ARCH-01 Critical, DOC-02 High).**
  `workflow-templates/python-slsa.yml` referenced the org reusable
  `.github/workflows/python-slsa.yml`, which exposes only `on: workflow_dispatch`
  (no `workflow_call`), so the template failed to load for any adopter. Now calls
  the SLSA generator directly, pinned to the same v2.1.0 SHA the reusable uses.
  Commit message was corrected from the original branch, which wrongly attributed
  the failure to "GitHub prohibits nested reusable calls" (GitHub allows 4 levels;
  the real cause was the missing `workflow_call` trigger).
- **`ci`: repo-local pre-commit lint gate (CICD-01).** New `pre-commit.yml` runs
  yamllint, markdownlint, the `no-em-dash` guard, and secret scans on every
  push/PR, closing the hole where these ran only as local hooks. Carries
  `permissions: {}` deny-by-default and RAD markers per repo convention. pre-commit
  is installed via the repo's locked `uv pip install --no-build 'pre-commit==4.6.0'`
  pattern (not bare pip).
- **`feat`: operator-script hardening + Bats coverage (CQ-02, CQ-01).**
  `transfer-repos.sh` gains `--dry-run` and surfaces `gh` stderr instead of
  discarding it; new Bats suites for `transfer-repos.sh` and `sync_org_files.sh`.
  All 8 new tests pass (network-free, gh/curl stubbed).

## Deliberately excluded

- **Audit docs** (`docs/audit/2026-05-29/`): not published. `docs/audit/` is now
  gitignored; the structured `findings.json` manifest is kept locally for
  remediation tracking.
- **`v1` versioning reconciliation** (DOC-01/DOC-06/SEC-04): dropped. It was built
  on the premise that `v1` is the moving tag tracking `main`. The remote actually
  has a full semver ladder `v1.0.0..v1.6.5` where `v1.6.5 == main` and `v1` is
  frozen 32 commits behind. The CHANGELOG narrative and the `@main`->`@v1` doc
  sweep would have been wrong. Needs redoing against the real ladder.
- **`workflow-templates` thin-caller README** (ARCH-02): dropped. It repeated the
  same incorrect "nested calls prohibited" claim.

## Validation

- `pre-commit run --all-files`: all hooks pass.
- New Bats suites: 8/8 pass via vendored bats-core.
- All PR CI checks pass (22/22, 4 conditional skips), including the new
  pre-commit gate running on its own branch, REUSE Compliance, and the
  **SonarCloud quality gate (0 open issues)**.
- Branch rebases cleanly on `main` (merge-base is current `main` HEAD).

## Resolved during review

- REUSE Compliance initially failed: a Bats assertion contained a literal
  `SPDX-License-Identifier:` token that the scanner read as an inline license
  declaration. Fixed with `REUSE-IgnoreStart/End` fences.
- SonarCloud initially failed `githubactions:S8544` on the bare `pip install`
  in the new gate. Fixed by locking the install via uv (`--no-build`,
  pinned version), not by suppression.

## Follow-up (separate work)

- Redo the versioning reconciliation against the `v1.6.x` ladder, and decide
  whether `v1` should be repointed to `v1.6.5` or retired.

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by williaby** [comment]
### PR Review

Automated multi-agent review (4 agents) plus GitHub Copilot. **CI: all 22 checks green; SonarCloud quality gate passed.** No Critical-tier findings. The items below are the Important tier; Suggested/Informational findings were also produced and are available in the review session.

#### Important (should fix)

1. **Knowingly-false code comment** [`workflow-templates/python-slsa.yml:71-74`](https://github.com/ByronWilliamsCPA/.github/blob/2935d467ea0af64ea1b9ea64e9d519b19e1f74ad/workflow-templates/python-slsa.yml#L71-L74): the added comment claims "GitHub Actions prohibits nested reusable workflow calls", which this PR's own description identifies as false. The real reason is that the org reusable `python-slsa.yml` exposes only `on: workflow_dispatch` (no `workflow_call`). Rewrite the comment to cite the missing trigger, not nesting depth. (Also flagged independently by Copilot.)

2. **Possible secret leak via `GH_DEBUG`** [`scripts/sync-secrets.sh:100-112`](https://github.com/ByronWilliamsCPA/.github/blob/2935d467ea0af64ea1b9ea64e9d519b19e1f74ad/scripts/sync-secrets.sh#L100-L112): the new `2>&1` capture is safe under normal `gh` behavior, but with `GH_DEBUG` exported, `gh` prints the request body (including the secret) to stderr, which the script now echoes. Add `unset GH_DEBUG GH_PAGER` before the loop.

3. **Untested core behavior** [`scripts/transfer-repos.sh:151-154`](https://github.com/ByronWilliamsCPA/.github/blob/2935d467ea0af64ea1b9ea64e9d519b19e1f74ad/scripts/transfer-repos.sh#L151-L154): the new error-surfacing failure branch has no test (the stub always succeeds), so a revert to `&>/dev/null` would go uncaught. Add a test that forces the transfer endpoint to exit non-zero.

4. **No test for sync-secrets error surfacing**: `scripts/sync-secrets.sh` gained the same error-surfacing change but has no Bats file at all; a `feat` PR should ship tests for new behavior.

5. **Comment contradicts code** [`.github/workflows/pre-commit.yml:16`](https://github.com/ByronWilliamsCPA/.github/blob/2935d467ea0af64ea1b9ea64e9d519b19e1f74ad/.github/workflows/pre-commit.yml#L16): header says "installed directly via pip" but the step uses `uv pip install --no-build`. The "via pip" phrasing invites reintroducing bare pip (fails SonarCloud S8544). Update the header.

6. **CHANGELOG not updated**: `feat:`/`fix:` commits with no CHANGELOG entry (OpenSSF baseline). Confirm against this repo's CHANGELOG cadence.

7. **Missing RAD markers** [`workflow-templates/python-slsa.yml:75-87`](https://github.com/ByronWilliamsCPA/.github/blob/2935d467ea0af64ea1b9ea64e9d519b19e1f74ad/workflow-templates/python-slsa.yml#L75-L87): the provenance block has `id-token: write` (OIDC) and calls an external reusable workflow, both mandatory RAD-marker categories per project CLAUDE.md, but carries no `#CRITICAL`/`#VERIFY`.

**Verified correct:** FIPS doc path fix, SLSA SHA pin (= v2.1.0), all actions SHA-pinned, pre-commit.yml deny-by-default permissions, no em-dashes, Conventional Commits.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: workflow-templates/python-slsa.yml [hunk 1]
```diff
@@ -67,14 +67,21 @@ jobs:
           path: dist/
 
   # Generate SLSA provenance
+  #
+  # The SLSA generator is called directly here, not via the org reusable
+  # python-slsa.yml. GitHub Actions prohibits nested reusable workflow calls,
+  # so a template cannot call python-slsa.yml (which itself calls the slsa
+  # generator). Calling the generator directly keeps the call one level deep.
```
[the remark this answers is not in the document]
Comment (williaby) [reply workflow-templates/python-slsa.yml]: Fixed in c57b794: the comment now states the real cause (the org reusable `python-slsa.yml` exposes only `on: workflow_dispatch`, no `workflow_call`, so it cannot be loaded via `uses:`) and notes that nesting depth is not the issue (GitHub permits four levels). Thanks for catching this.

File: scripts/transfer-repos.sh [hunk 2]
```diff
@@ -122,6 +151,7 @@ for repo in "${REPO_ARRAY[@]}"; do
         fi
     else
         echo -e "${RED}❌${NC}"
+        echo "      ${transfer_output}"
```
[the remark this answers is not in the document]
Comment (williaby) [reply scripts/transfer-repos.sh]: Fixed in 873bc29: replaced `echo` with `printf '%s\n'` so the captured `gh` error renders exactly.

File: scripts/sync-secrets.sh [hunk 3]
```diff
@@ -100,13 +100,16 @@ for repo in "${REPO_ARRAY[@]}"; do
...
+    # passed via --body and is not echoed by gh on error.
+    if secret_output=$(gh secret set "$SECRET_NAME" \
         --repo "$FULL_REPO" \
-        --body "$SECRET_VALUE" 2>/dev/null; then
+        --body "$SECRET_VALUE" 2>&1); then
         echo -e "${GREEN}✅${NC}"
         SUCCESS=$((SUCCESS + 1))
     else
         echo -e "${RED}❌${NC}"
+        echo "      ${secret_output}"
```
[the remark this answers is not in the document]
Comment (williaby) [reply scripts/sync-secrets.sh]: Fixed in 873bc29: switched to `printf '%s\n'`. Also added an `unset GH_DEBUG GH_PAGER` guard before the loop so the captured stderr can never include the secret body under debug mode.

**Comment by williaby** [comment]
### PR Fix Summary

Addressed the review findings in 5 commits (c57b794..06508ce), all signed, pre-commit + commitizen green locally, 84/84 Bats tests pass.

**Fixed**
- `fix(slsa)` c57b794: rewrote the false "nested reusable calls prohibited" comment to the real cause (missing `workflow_call` trigger); scoped the `build` job to `permissions: {contents: read}` so it no longer inherits `id-token: write`; added RAD markers on the provenance job (OIDC + external caller).
- `fix(scripts)` 873bc29: `unset GH_DEBUG GH_PAGER` before the secret loop so a secret cannot leak into surfaced errors; `printf` instead of `echo` for captured `gh` output in both scripts.
- `docs(ci)` 10151e4: corrected the `pre-commit.yml` "installed via pip" header to the `uv pip install --no-build` reality; completed the `#VERIFY` RAD pairings.

**Added**
- `test(scripts)` 03376e1: transfer-failure error-surfacing test; new `tests/sync-secrets.bats` (auth guard, empty-name guard, happy path, failure path with a `refute_output` assertion proving the secret value never leaks).
- `docs(changelog)` 06508ce: `[Unreleased]` entries for the SLSA fix, FIPS doc path, pre-commit gate, and operator-script hardening.

**Deferred / not changed (with reasons)**
- `--dry-run` org-name prompt: by design. The dry-run message reports "would be transferred to <org>", so the target org is required input.
- `egress-policy: audit` -> `block`: needs a tested `allowed-endpoints` allowlist; flipping to `block` alone risks `startup_failure` (same class as the #136 revert). Tracked for a follow-up.
- `sha256sum *` (SC2035) in `python-slsa.yml` build hash step: pre-existing, untouched by this PR, and CI actionlint already passes on it. Worth a separate `sha256sum ./*` cleanup PR.
- Additional test depth (success-transfer path, sync-org partial-write, stub flag fidelity): non-blocking test debt.

CI re-run triggered by the push.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
